cmd/dataset/app/dataset.go

```diff
@@ -232,21 +236,22 @@ func handle() {
 	}
 }
 
-func NewCache(scheme *runtime.Scheme) cache.NewCacheFunc {
-	selectors := make(cache.SelectorsByObject, 1)
+func NewCacheOptions() cache.Options {
+	var cronJobKey client.Object
 
 	if compatibility.IsBatchV1CronJobSupported() {
-		selectors[&batchv1.CronJob{}] = cache.ObjectSelector{Label: labels.SelectorFromSet(labels.Set{
-			common.JobPolicy: common.CronPolicy,
-		})}
+		cronJobKey = &batchv1.CronJob{}
 	} else {
-		selectors[&batchv1beta1.CronJob{}] = cache.ObjectSelector{Label: labels.SelectorFromSet(labels.Set{
-			common.JobPolicy: common.CronPolicy,
-		})}
+		cronJobKey = &batchv1beta1.CronJob{}
 	}
 
-	return cache.BuilderWithOptions(cache.Options{
-		Scheme:            scheme,
-		SelectorsByObject: selectors,
-	})
+	return cache.Options{
+		ByObject: map[client.Object]cache.ByObject{
```

Comment:
Do we need to add scheme?

```golang
return cache.Options{
		Scheme: scheme,
		ByObject: map[client.Object]cache.ByObject{
			cronJobKey: {
				Label: labels.SelectorFromSet(labels.Set{
					common.JobPolicy: common.CronPolicy,
				}),
			},
		},
	}
```

Reply:
no need to, as the `Scheme` field in `ctrl.Options`  will used to create the default Client and Cache.

Reply:
How about setting scheme in cache option to keep consistency with previous version? My thought is to make it easy for maintenance.  WDYT？